lib/puppet/util/windows/adsi.rb

```diff
@@ -436,6 +436,28 @@ def unset_userflags(*flags)
       op_userflags(*flags) { |userflags, flag| userflags & ~ADS_USERFLAGS[flag] }
     end
 
+    def disabled?
+      userflag_set?(:ADS_UF_ACCOUNTDISABLE)
+    end
+
+    def locked_out?
+      # Note that the LOCKOUT flag is known to be inaccurate when using the
+      # LDAP IADsUser provider, but this class consistently uses the WinNT
+      # provider, which is expected to be accurate.
+      userflag_set?(:ADS_UF_LOCKOUT)
+    end
+
+    def expired?
```

Comment:
Nit: Could be written as:
```
def expired?
  expires = ...
  expires && ..
rescue WIN32OLERuntimeError => e
  raise e unless ..
  false
end
```

This would avoid the `begin` block.

Reply:
ok